File: packages/sdk-java/src/main/java/org/melaya/MfaAPI.java
package org.melaya;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * MFA (Multi-Factor Authentication) API — enroll, confirm, and check TOTP.
 *
 * <p>Maps to {@code /api/v1/private/mfa/*}.
 */
public class MfaAPI {

    private final HttpClient http;

    MfaAPI(HttpClient http) {
        this.http = http;
    }

    /** Return the caller's MFA enrollment status. */
    public JsonNode status() {
        return http.get("/api/v1/private/mfa/status", null);
    }

    /**
     * Initiate TOTP setup. Returns a QR-code URI and raw secret.
     * The caller must display the QR code and then confirm with {@link #confirmSetup}.
     */
    public JsonNode setup() {
        return http.post("/api/v1/private/mfa/setup", null);
    }

    /**
     * Confirm TOTP setup with the first valid code; activates MFA on the account.
     *
     * @param code the first TOTP code from the authenticator app
     */
    public JsonNode confirmSetup(String code) {
        return http.post("/api/v1/private/mfa/confirm",
                MarketAPI.params("code", code));
    }
}

File: packages/sdk-kotlin/src/main/kotlin/org/melaya/MfaAPI.kt
package org.melaya

import org.json.JSONObject

/**
 * MFA API — enroll and manage TOTP multi-factor authentication.
 *
 * Paths:
 *   - `GET  /api/v1/private/mfa/status`  — MFA enrollment status
 *   - `POST /api/v1/private/mfa/setup`   — initiate TOTP setup; returns QR/secret
 *   - `POST /api/v1/private/mfa/confirm` — confirm TOTP setup with first valid code
 */
class MfaAPI internal constructor(private val http: HttpClient) {

    /** Return MFA enrollment status for the caller. */
    fun status(): JSONObject {
        return http.get("/api/v1/private/mfa/status").asObject()
    }

    /**
     * Initiate TOTP setup.
     * Returns a QR code URL and/or the raw TOTP secret to add to an authenticator app.
     */
    fun setup(): JSONObject {
        return http.post("/api/v1/private/mfa/setup").asObject()
    }

    /**
     * Confirm TOTP setup with the first valid [code] from the authenticator app.
     * Activates MFA on the account.
     */
    fun confirmSetup(code: String): JSONObject {
        return http.post(
            "/api/v1/private/mfa/confirm",
            mapOf("code" to code)
        ).asObject()
    }
}
